### PR TITLE
Update projects controller

### PR DESCRIPTION
**Pull request by SerjVankovich**

### CODE REVIEW

File: src/main/java/org/spburegistry/backend/service/implementation/ProjectServiceImpl.java [hunk 1]
```diff
@@ -106,6 +103,93 @@ public Iterable<ProjectTO> findRandomProjects(Optional<Long> limit) {
...
+        updateProjectStudents(project, projectRequestTO.getProjectRoles());
+        updateLinkingProjectIds(project, projectRequestTO.getLinkedProjectsIds());
+
+        updateProjectLinks(project, projectRequestTO.getLinks());
+        updateProjectRoles(project, projectRequestTO.getProjectRoles());
+        return ConvertToTO.projectToTO(projectRepo.save(project));
+    }
+
+
+    private void updateLinkingProjectIds(Project project, Set<Long> linkedProjectsIds) {
```
Comment (End-rey) [inline src/main/java/org/spburegistry/backend/service/implementation/ProjectServiceImpl.java:141]: why

File: src/main/java/org/spburegistry/backend/service/implementation/ProjectServiceImpl.java [hunk 2]
```diff
@@ -106,6 +103,93 @@ public Iterable<ProjectTO> findRandomProjects(Optional<Long> limit) {
...
+        project.setLinkedProjects(getLinkedProjects(linkedProjectsIds));
+    }
+
+    private void updateProjectTags(Project project, Set<TagTO> tags) {
```
Comment (End-rey) [inline src/main/java/org/spburegistry/backend/service/implementation/ProjectServiceImpl.java:145]: why

File: src/main/java/org/spburegistry/backend/service/implementation/ProjectServiceImpl.java [hunk 3]
```diff
@@ -106,6 +103,93 @@ public Iterable<ProjectTO> findRandomProjects(Optional<Long> limit) {
...
+        project.setTags(getTags(tags));
+    }
+
+    private void updateProjectClinics(Project project, Set<Long> clinicsIds) {
```
Comment (End-rey) [inline src/main/java/org/spburegistry/backend/service/implementation/ProjectServiceImpl.java:149]: why

File: src/main/java/org/spburegistry/backend/service/implementation/ProjectServiceImpl.java [hunk 4]
```diff
@@ -106,6 +103,93 @@ public Iterable<ProjectTO> findRandomProjects(Optional<Long> limit) {
...
+        project.setClinics(getClinics(clinicsIds));
+    }
+
+    private void updateProjectClients(Project project, Set<Long> clientsIds) {
```
Comment (End-rey) [inline src/main/java/org/spburegistry/backend/service/implementation/ProjectServiceImpl.java:153]: why

File: src/main/java/org/spburegistry/backend/service/implementation/ProjectServiceImpl.java [hunk 5]
```diff
@@ -106,6 +103,93 @@ public Iterable<ProjectTO> findRandomProjects(Optional<Long> limit) {
...
+        project.setClients(getClients(clientsIds));
+    }
+
+    private void updateProjectCurators(Project project, Set<Long> curatorsIds) {
```
Comment (End-rey) [inline src/main/java/org/spburegistry/backend/service/implementation/ProjectServiceImpl.java:157]: why

File: src/main/java/org/spburegistry/backend/service/implementation/ProjectServiceImpl.java [hunk 6]
```diff
@@ -106,6 +103,93 @@ public Iterable<ProjectTO> findRandomProjects(Optional<Long> limit) {
...
+        project.setCurators(getCurators(curatorsIds));
+    }
+
+    private void updateProjectSupervisors(Project project, Set<Long> supervisorsIds) {
```
Comment (End-rey) [inline src/main/java/org/spburegistry/backend/service/implementation/ProjectServiceImpl.java:161]: why

File: src/main/java/org/spburegistry/backend/service/implementation/ProjectServiceImpl.java [hunk 7]
```diff
@@ -106,6 +103,93 @@ public Iterable<ProjectTO> findRandomProjects(Optional<Long> limit) {
...
+        project.setSupervisors(getSupervisors(supervisorsIds));
+    }
+
+    private void updateProjectStudents(Project project, Set<RoleTO> projectRoles) {
```
Comment (End-rey) [inline src/main/java/org/spburegistry/backend/service/implementation/ProjectServiceImpl.java:165]: why

**Review by End-rey — APPROVED** [review]
refactor code, but everything else is good

**Review by MicAPic — APPROVED** [review]

**Pull request merged**